app/src/main/java/com/lastaoutdoor/lasta/data/db/ActivitiesDBRepositoryImpl.kt

```diff
@@ -156,4 +156,64 @@ constructor(context: Context, database: FirebaseFirestore) : ActivitiesDBReposit
     document.update("numRatings", FieldValue.increment(1))
     document.update("rating", newMeanRating)
   }
+
+  override suspend fun deleteRating(activity: Activity, userId: String): Activity {
+    val document = activitiesCollection.document(activity.activityId)
+
+    // Fetch the current ratings from the activity document
+    val activitySnapshot = document.get().await()
+    val currentRatings = activitySnapshot.get("ratings") as? List<Map<String, Any>> ?: emptyList()
+
+    // Find the rating to be removed
+    val ratingToRemove = currentRatings.find { it["userId"] == userId }
+
+    if (ratingToRemove != null) {
+      // Update the document to remove the rating
+      document.update("ratings", FieldValue.arrayRemove(ratingToRemove)).await()
+
+      // Recalculate the number of ratings
+      val newNumRatings = activity.numRatings - 1
+
+      // Recalculate the new average rating
+      val newRatings = currentRatings.filter { it["userId"] != userId }
+      val newMeanRating =
+          if (newNumRatings > 0) {
+            newRatings.map { it["rating"].toString().toFloat() }.average()
+          } else {
+            1.0

```

Comment:
Yes I agree with that, but it is implemented like this elsewhere in the code, I just wanted to be consistent with it! What should I do?

Reply:
Yes it was already implemented that way, but still I don't find it very intuitive. 
I think it would be best to just leave it as it is for the moment and discuss it altogether tomorrow and make it a small task for next week if needed.

Thanks for your reply